src/order/compactly_generated.lean

```diff
@@ -223,12 +218,23 @@ alias is_Sup_finite_compact_iff_is_sup_closed_compact ↔
       _ is_sup_closed_compact.is_Sup_finite_compact
 alias is_sup_closed_compact_iff_well_founded ↔ _ well_founded.is_sup_closed_compact
 
+end complete_lattice
+
+/-- A complete lattice is said to be compactly generated if any
+element is the `Sup` of compact elements. -/
+class is_compactly_generated (α : Type*) [complete_lattice α] : Prop :=
+(exists_Sup_eq :
+  ∀ (x : α), ∃ (s : set α), (∀ x ∈ s, complete_lattice.is_compact_element x) ∧ Sup s = x)
+
+namespace complete_lattice
+variables {α : Type*} [complete_lattice α]
+
 lemma compactly_generated_of_well_founded (h : well_founded ((>) : α → α → Prop)) :
```

Comment:
Should we have `instance [has_well_founded α] : is_compactly_generated α` after this `lemma` as well?

Reply:
`has_well_founded` only means there is some canonical relation on the given type, not that `<=` itself is well-founded. We still need to make a class for well-founded orders, but we've gotten stuck repeatedly in bikeshedding.

Reply:
I might be missing something, but doesn't `has_well_founded.wf` contain a proof that the relation `has_well_founded.r` is well-founded? https://leanprover-community.github.io/mathlib_docs/init/wf.html#has_well_founded

Reply:
Yes, but `has_well_founded.r` doesn't actually have to be the same relation as `has_le.le`

Reply:
Ah, got it. Thanks!